### PR TITLE
ci: use github action for bulk of github test

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

We now have the lovely [github-action-add-on-test](https://github.com/ddev/github-action-add-on-test) from @julienloizelet - THANKS!

This uses it.

### CODE REVIEW

**Comment by rfay** [comment]
Please confirm correctness @julienloizelet @tyler36 

I also implemented this in [ddev-platformsh](https://github.com/ddev/ddev-platformsh)

**Comment by julienloizelet** [comment]
Thanks @rfay ,

 - We could remove the `defaults` part : https://github.com/ddev/ddev-addon-template/blob/f1d038d8ed87bf48baa7ce0998ce5343ec0ac36e/.github/workflows/tests.yml#L28-L30

We already set bash as shell in every step of the action because it is required to define one : https://github.com/ddev/github-action-add-on-test/blob/main/action.yaml#L63 for example

- We could remove the `env` part https://github.com/ddev/ddev-addon-template/blob/f1d038d8ed87bf48baa7ce0998ce5343ec0ac36e/.github/workflows/tests.yml#L22-L24

The DDEV_GITHUB_TOKEN value is already set in the action: https://github.com/ddev/github-action-add-on-test/blob/main/action.yaml#L98-L100

There's no harm in keeping it, but :

- the fewer things there are outside the action, the less maintenance is required
- If someone set the DDEV_GITHUB_TOKEN value outside the action, it will be overridden in the action by the `token` input value.
- Everyday, this run https://github.com/ddev/github-action-add-on-test/actions/workflows/add-ons-test.yml launches about 20 quasi simultaneous `ddev get` without reaching the API rate limit (It doesn't prove anything, but it suggests that it works.).

File: README.md [hunk 1]
```diff
@@ -40,12 +40,10 @@ A repository like this one is the way to get started. You can create a new repo
 8. Create a release on GitHub.
 9. Test manually with `ddev get <owner/repo>`.
 10. You can test PRs with `ddev get https://github.com/<user>/<repo>/tarball/<branch>`
-11. Update the README.md to describe the add-on, how to use it, and how to contribute. If there are any manual actions that have to be taken, please explain them. If it requires special configuration of the using project, please explain how to do those. Examples in [ddev/ddev-drupal9-solr](https://github.com/ddev/ddev-drupal9-solr), [ddev/ddev-memcached](github.com/ddev/ddev-memcached), and [ddev/ddev-beanstalkd](https://github.com/ddev/ddev-beanstalkd).
+11. Update the README.md to describe the add-on, how to use it, and how to contribute. If there are any manual actions that have to be taken, please explain them. If it requires special configuration of the using project, please explain how to do those. Examples in [ddev/ddev-solr](https://github.com/ddev/ddev-solr), [ddev/ddev-memcached](github.com/ddev/ddev-memcached), and (advanced) [ddev-platformsh](https://github.com/ddev/ddev-platformsh).
```
Comment (tyler36) [inline README.md:43]:
```suggestion
11. Update the README.md to describe the add-on, how to use it, and how to contribute. If there are any manual actions that have to be taken, please explain them. If it requires special configuration of the using project, please explain how to do those. Examples in [ddev/ddev-solr](https://github.com/ddev/ddev-solr), [ddev/ddev-memcached](github.com/ddev/ddev-memcached), and [ddev-platformsh](https://github.com/ddev/ddev-platformsh).
```

Not sure if you want to include advance or it is left over

**Review by tyler36 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (rfay) [reply README.md:43]: It was intentional, it's a hard one to understand, but demonstrates the most exotic things I know to do with an add-on. Probably too exotic.

**Comment by rfay** [comment]
See what you think now @julienloizelet - thanks for paying careful attention.

**Review by tyler36 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (tyler36) [reply README.md:43]: > It was intentional

Thank you for clarifying. Looks good to me.

**Pull request merged**